api/graphql/reservations/reservation_serializers/staff_create_serializers.py

```diff
@@ -0,0 +1,228 @@
+        self.fields["confirmed_at"].read_only = True
+        self.fields["handled_at"].read_only = True
+        self.fields["begin"].required = True
+        self.fields["end"].required = True
+
+        # Optional fields
+        self.fields["billing_first_name"].required = False
+        self.fields["billing_last_name"].required = False
+        self.fields["billing_phone"].required = False
+        self.fields["billing_email"].required = False
+        self.fields["billing_address_street"].required = False
+        self.fields["billing_address_city"].required = False
+        self.fields["billing_address_zip"].required = False
+        self.fields["reservee_type"].required = False
+        self.fields["reservee_first_name"].required = False
+        self.fields["reservee_last_name"].required = False
+        self.fields["reservee_organisation_name"].required = False
+        self.fields["reservee_phone"].required = False
+        self.fields["reservee_email"].required = False
+        self.fields["reservee_id"].required = False
+        self.fields["reservee_is_unregistered_association"].required = False
+        self.fields["reservee_address_street"].required = False
+        self.fields["reservee_address_city"].required = False
+        self.fields["reservee_address_zip"].required = False
+        self.fields["home_city_pk"].required = False
+        self.fields["age_group_pk"].required = False
+        self.fields["applying_for_free_of_charge"].required = False
+        self.fields["free_of_charge_reason"].required = False
+        self.fields["name"].required = False
+        self.fields["description"].required = False
+        self.fields["num_persons"].required = False
+        self.fields["purpose_pk"].required = False
+        self.fields["buffer_time_after"].required = False
+        self.fields["buffer_time_before"].required = False
+
+    def validate(self, data):
+        data = super().validate(data)
+
+        begin = data.get("begin").astimezone(DEFAULT_TIMEZONE)
+        end = data.get("end").astimezone(DEFAULT_TIMEZONE)
+
+        reservation_units = data.get(
+            "reservation_unit",
+        )
+
+        self.check_begin(begin, end)
+
+        for reservation_unit in reservation_units:
+            self.check_reservation_overlap(reservation_unit, begin, end)
+            self.check_buffer_times(reservation_unit, begin, end)
+            self.check_reservation_intervals_for_staff_reservation(
+                begin, reservation_unit
+            )
+
+        now = datetime.datetime.now(tz=DEFAULT_TIMEZONE)
+        data["handled_at"] = now
+        data["confirmed_at"] = now
+        data["state"] = ReservationState.CONFIRMED
+
+        self.set_buffer_times(data)
+
+        user = self.context.get("request").user
+        data["user"] = user
+
+        return data
+
+    def set_buffer_times(self, data):
+        if not data.get("buffer_time_after"):
+            data["buffer_time_after"] = None

```

Comment:
Could you clarify what is the purpose of this? Initially it looked like you are trying to set `buffer_time_after` to `None` if it is `None`, but perhaps the point here is to replace other falsy values like empty strings with `None`?

Reply:
Yup, no point, I was so heavily comparing the code to the "regular" create where the buffer time gets copied from reservation_unit. I'll change this. Good catch 👍

Reply:
its now removed 🧘‍♂️